crates/ruma-common/src/push/iter.rs

```diff
@@ -114,7 +115,9 @@ impl IntoIterator for Ruleset {
 
 /// Reference to any kind of push rule.
 #[derive(Clone, Copy, Debug)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
 pub enum AnyPushRuleRef<'a> {
+    // TODO: is non_exhaustive useful here?
```

Comment:
Here and for `AnyPushRule`, does it make sense to have them non_exhaustive, or are they very unlikely to change?

Reply:
That are very unlikely to change because the current notification system is likely getting replaced entirely. Then again it probably doesn't hurt to make these non-exhaustive (with the usual opt-out).